src/stirling/source_connectors/file_source/file_source_connector.cc

```diff
@@ -168,6 +200,101 @@ Status FileSourceConnector::StopImpl() {
 
 constexpr int kMaxLines = 1000;
 
+void FileSourceConnector::TransferDataFromTetragon(DataTable::DynamicRecordBuilder* /*r*/,
+                                                 const std::string& line) {
+  DataTable::DynamicRecordBuilder r(data_tables_[0]);
+  rapidjson::Document doc;
+  doc.Parse(line.c_str());
+
+  if (doc.HasParseError() || !doc.IsObject()) {
+    LOG(ERROR) << "Invalid JSON line: " << line;
+    return;
+  }
+
+  // Extract "time"
+  std::string timeStr = "empty";
+  if (doc.HasMember("time") && doc["time"].IsString()) {
+    timeStr = doc["time"].GetString();
+  }
+  r.Append(0, types::StringValue(timeStr), kMaxStringBytes);
+
+  // Extract "node_name"
+  std::string nodeNameStr = "empty";
+  if (doc.HasMember("node_name") && doc["node_name"].IsString()) {
+    nodeNameStr = doc["node_name"].GetString();
+  }
+  r.Append(1, types::StringValue(nodeNameStr), kMaxStringBytes);
```

Comment:
I think we should consider how to extend the `DynamicRecordBuilder` to have default values for columns. It would be nice if the schema defined if a column had a default value and could handle this case without the `TransferData` function needing to specially handle specific columns.

I don't think this should block you from moving forward with this depending on the priorities, but I think it would be possible to enhance `TransferDataFromJSON` to handle this use case.

Reply:
@entlein